src/NCronJob/Registry/IInstantJobRegistry.cs

```diff
@@ -151,37 +190,65 @@ public void RunScheduledJob<TJob>(DateTimeOffset startDate, object? parameter =
+    public void RunScheduledJob(Delegate jobDelegate, TimeSpan delay, CancellationToken token = default)
+    {
+        var utcNow = timeProvider.GetUtcNow();
+        RunScheduledJob(jobDelegate, utcNow + delay, token);
+    }
+
+    /// <inheritdoc />
+    public void RunScheduledJob(Delegate jobDelegate, DateTimeOffset startDate, CancellationToken token = default) =>
+        RunDelegateJob(jobDelegate, startDate, false, token);
+
+    /// <inheritdoc />
+    public void ForceRunScheduledJob<TJob>(TimeSpan delay, object? parameter = null, CancellationToken token = default)
+        where TJob : IJob
     {
         var utcNow = timeProvider.GetUtcNow();
-        RunScheduledJob(jobDelegate, utcNow + delay);
+        RunJob<TJob>(utcNow + delay, parameter, true, token);
     }
 
     /// <inheritdoc />
-    public void RunScheduledJob(Delegate jobDelegate, DateTimeOffset startDate)
+    public void ForceRunScheduledJob(Delegate jobDelegate, TimeSpan delay, CancellationToken token = default)
+    {
+        var utcNow = timeProvider.GetUtcNow();
+        ForceRunScheduledJob(jobDelegate, utcNow + delay, token);
+    }
+
+    /// <inheritdoc />
+    public void ForceRunScheduledJob(Delegate jobDelegate, DateTimeOffset startDate, CancellationToken token = default) =>
+        RunDelegateJob(jobDelegate, startDate, true, token);
+    
+    /// <inheritdoc />
+    public void ForceRunInstantJob(Delegate jobDelegate, CancellationToken token = default) =>
+        ForceRunScheduledJob(jobDelegate, TimeSpan.Zero, token);
+
+    /// <inheritdoc />
+    public void ForceRunInstantJob<TJob>(object? parameter = null, CancellationToken token = default)
+        where TJob : IJob => ForceRunScheduledJob<TJob>(TimeSpan.Zero, parameter, token);
+    
+    private void RunDelegateJob(Delegate jobDelegate, DateTimeOffset startDate, bool forceExecution = false, CancellationToken token = default)
     {
         var definition = dynamicJobFactoryRegistry.Add(jobDelegate);
         var run = JobRun.Create(definition);
         run.Priority = JobPriority.High;
         run.RunAt = startDate;
         run.IsOneTimeJob = true;
+        run.CancellationToken = token;
 
         var jobQueue = jobQueueManager.GetOrAddQueue(run.JobDefinition.JobFullName);
-        jobQueue.EnqueueForDirectExecution(run, startDate);
-    }
 
-    /// <inheritdoc />
-    public void ForceRunInstantJob<TJob>(object? parameter = null, CancellationToken token = default)
-        where TJob : IJob => ForceRunInstantJob<TJob>(TimeSpan.Zero, parameter, token);
-    
-    /// <inheritdoc />
-    public void ForceRunInstantJob<TJob>(TimeSpan delay, object? parameter = null, CancellationToken token = default)
-        where TJob : IJob
-    {
-        var utcNow = timeProvider.GetUtcNow();
-        RunJob<TJob>(utcNow + delay, parameter, true,token);
+        if (forceExecution)
+        {
+            _ = jobProcessor.ProcessJobAsync(run, token);
```

Comment:
With regard to the UI/Dashboard - will "forced instant jobs" be listed if we bypass the whole queue?

Reply:
Good question! I think the dashboard needs two visual areas (maybe more): one that represents all the discoverable jobs, so the jobs registry. The other would be for the Job related queues with the ability to drill down into the instances of each job definition. A forced instant job will not show in a queue because it just bypasses it, but should show in any real-time stream and stored audit for that Job Type. I'll mock something up to help visualize, either in HTML or Figma, before committing to the UX.